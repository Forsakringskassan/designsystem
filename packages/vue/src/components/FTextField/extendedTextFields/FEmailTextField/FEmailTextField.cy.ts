import { FTextFieldPageObject } from "../../../../cypress/FTextField.pageobject";
import FEmailTextField from "./FEmailTextField.vue";

function mount(): FTextFieldPageObject {
    cy.mount(FEmailTextField, {
        attrs: { "extended-validation": true },
        slots: {
            extendedLabel: "",
        },
    });
    return new FTextFieldPageObject("body");
}

describe("FEmailTextField", () => {
    const selector = "body";
    const firstEmailTextField = new FTextFieldPageObject(
        `${selector} .text-field:nth(0)`,
    );
    const secondEmailTextField = new FTextFieldPageObject(
        `${selector} .text-field:nth(1)`,
    );

    beforeEach(() => {
        mount();
    });

    describe("extended validation", () => {
        it("should have two email fields with the second matching the input of the first", () => {
            firstEmailTextField.label.errorIcon().should("not.exist");
            secondEmailTextField.label.errorIcon().should("not.exist");
            firstEmailTextField
                .input()
                .focus()
                .clear()
                .type("user@example.net");
            secondEmailTextField.input().click();
            firstEmailTextField.label.errorIcon().should("not.exist");
            secondEmailTextField.label.errorIcon().should("not.exist");

            secondEmailTextField
                .input()
                .focus()
                .clear()
                .type("user@example.net");

            firstEmailTextField.input().click();
            firstEmailTextField.label.errorIcon().should("not.exist");
            secondEmailTextField.label.errorIcon().should("not.exist");
        });

        it("should have two email fields with the second matching displaying error if it doesnt match input of the first", () => {
            firstEmailTextField
                .input()
                .focus()
                .clear()
                .type("user@example.net");

            secondEmailTextField
                .input()
                .focus()
                .clear()
                .type("anotheruser@example.net");

            firstEmailTextField.input().click();
            firstEmailTextField.label.errorIcon().should("not.exist");
            secondEmailTextField.label.errorIcon().should("be.visible");
        });

        it("[KNOWN BUG]: should have two email fields with the second matching displaying error if first is updated", () => {
            firstEmailTextField
                .input()
                .focus()
                .clear()
                .type("user@example.net");

            secondEmailTextField
                .input()
                .focus()
                .clear()
                .type("user@example.net");

            firstEmailTextField.input().click();
            firstEmailTextField.label.errorIcon().should("not.exist");
            secondEmailTextField.label.errorIcon().should("not.exist");

            firstEmailTextField
                .input()
                .clear()
                .focus()
                .clear()
                .type("anotheruser@example.net");

            secondEmailTextField.input().click();
            firstEmailTextField.label.errorIcon().should("not.exist");

            /*
             * [KNOWN BUG]: SFKUI-1858 - "Matches validator är inte responsiv för bevakade fältet"
             *
             * Enable:a när buggen är fixad.
             */
            /*
            secondEmailTextField.label.errorIcon().should('be.visible');
            */
        });

        it("should have two email fields with the second matching displaying error if second is updated", () => {
            firstEmailTextField
                .input()
                .focus()
                .clear()
                .type("user@example.net");

            secondEmailTextField
                .input()
                .focus()
                .clear()
                .type("user@example.net");

            firstEmailTextField.input().click();
            firstEmailTextField.label.errorIcon().should("not.exist");
            secondEmailTextField.label.errorIcon().should("not.exist");

            secondEmailTextField
                .input()
                .clear()
                .focus()
                .clear()
                .type("anotheruser@example.net");

            firstEmailTextField.input().click();
            firstEmailTextField.label.errorIcon().should("not.exist");
            secondEmailTextField.label.errorIcon().should("be.visible");
        });
    });
});
