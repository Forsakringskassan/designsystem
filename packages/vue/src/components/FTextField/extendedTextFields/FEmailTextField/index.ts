export { default as FEmailTextField } from "./FEmailTextField.vue";
