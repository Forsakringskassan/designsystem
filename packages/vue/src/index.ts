/* include typescript declarations for vite static asset handling (e.g. `?raw`) */
/// <reference types="vite/client" />

export * from "./components";
export {
    type UseSlotUtils,
    type WithoutInstance,
    useSlotUtils,
} from "./composables";
export {
    config,
    type FKUIConfig,
    FKUIConfigButtonOrder,
    type FKUIContext,
    type MaybeWithFKUIContext,
    setRunningContext,
} from "./config";
export * from "./internal-components";
export * from "./plugins";
export {
    type ComponentUnmountEvent,
    type ComponentValidityEvent,
    type ErrorItem,
    type GroupValidityEvent,
    type ListArray,
    type ListItem,
    type UnknownItem,
    ErrorData,
    ErrorViewData,
    FormErrorList,
    MenuAction,
} from "./types";
export * from "./utils";
