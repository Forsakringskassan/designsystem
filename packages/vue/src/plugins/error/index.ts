export {
    type ErrorPluginOptions,
    default as ErrorPlugin,
    UNHANDLED_ERROR_EVENT,
} from "./ErrorPlugin";
export { default as FErrorHandlingApp } from "./FErrorHandlingApp.vue";
