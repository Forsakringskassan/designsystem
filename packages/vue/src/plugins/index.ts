export * from "./test";
export * from "./translation";
export * from "./validation";
export * from "./error";
export { FormatPlugin } from "./format";
