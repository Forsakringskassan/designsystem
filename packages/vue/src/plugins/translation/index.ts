/* istanbul ignore file */
export { type TranslateFunction } from "./translate-function";
export { TranslationMixin, TranslationPlugin } from "./TranslationPlugin";
export { useTranslate } from "./use-translate";
