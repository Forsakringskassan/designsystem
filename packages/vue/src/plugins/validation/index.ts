/* istanbul ignore file */
export { ValidationPlugin } from "./ValidationPlugin";
