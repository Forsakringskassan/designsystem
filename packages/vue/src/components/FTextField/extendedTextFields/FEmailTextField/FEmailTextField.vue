<script lang="ts">
import { defineComponent } from "vue";
import {
    type ValidatorConfigs,
    type ValidityEvent,
    ElementIdService,
    ValidationService,
    TranslationService,
} from "@fkui/logic";
import { FTextField } from "../..";
import { TranslationMixin } from "../../../../plugins";
import { renderSlotText, dispatchComponentValidityEvent } from "../../../../utils";

export default defineComponent({
    name: "FEmailTextField",
    components: { FTextField },
    mixins: [TranslationMixin],
    inheritAttrs: false,
    props: {
        /**
         * The id for the input id attribute.
         * The id for the label for attribute.
         * If the prop is not set a random value will be generated.
         */
        id: {
            type: String,
            required: false,
            default: () => ElementIdService.generateElementId(),
        },
        /**
         * The value for the input.
         * If the prop is not set undefined will be used.
         * @model
         */
        modelValue: {
            type: [String, null],
            required: false,
            default: undefined,
        },
        maxLength: {
            type: Number,
            default: 80,
        },
        extendedValidation: { type: Boolean, default: false },
        /**
         * The error message to be displayed on paste
         * If the prop is not set the default text "Du kan inte kopiera mejladressen. Du måste skriva in den igen." will be set
         */
        pasteErrorText: {
            type: String,
            required: false,
            default: TranslationService.provider.translate(
                "fkui.email-text-field.error.pasting",
                "Du kan inte kopiera mejladressen. Du måste skriva in den igen.",
            ),
        },
    },
    emits: ["blur", "change", "update:modelValue"],
    data() {
        return {
            validityMode: "INITIAL" as string,
            secondEmail: "",
            showPasteErrorMessage: false,
            defaultText: this.$t("fkui.email-text-field.label", "Mejladress"),
        };
    },
    mounted() {
        this.configureValidators();
    },
    methods: {
        onChange(event: Event): void {
            /**
             * @event change
             * @type {string | number}
             */
            this.$emit("change", event);
        },
        onBlur(event: Event): void {
            /**
             * @event blur
             * @type {string | number}
             */
            this.$emit("blur", event);
        },
        onUpdate(event: Event): void {
            /**
             * V-model event.
             *
             * @event update:modelValue
             * @type {string}
             */
            this.$emit("update:modelValue", event);
        },
        onPaste(event: Event): boolean {
            this.showPasteErrorMessage = true;
            event.preventDefault();
            return false;
        },
        onValidity({ detail }: CustomEvent<ValidityEvent>): void {
            this.showPasteErrorMessage = false;
            this.validityMode = detail.validityMode;
            const errorMessage = renderSlotText(this.$slots.default) ?? this.defaultText;
            const element = this.$el.querySelector(`#${detail.elementId}`);
            if (element) {
                dispatchComponentValidityEvent(element, {
                    ...detail,
                    errorMessage,
                    focusElementId: detail.elementId,
                });
            }
        },
        onPendingValidity(): void {
            this.validityMode = "INITIAL";
        },
        configureValidators(): void {
            const elements: HTMLInputElement[] = Array.from(this.$el.querySelectorAll("input"));
            const validatorEmailConfig: ValidatorConfigs = {
                email: {},
            };
            /*
             * Since this is an extended component the third argument "isBaseConfigs" for addValidatorsToElement() is set to true
             * in order to keep the base configuration set in this constructor when the application code adds its own validators, for example:
             * <f-email-text-field v-validation.required>
             */
            ValidationService.addValidatorsToElement(elements[0], validatorEmailConfig, true);

            if (this.extendedValidation) {
                this.configureExtendedValidation(elements);
            }
        },
        configureExtendedValidation(elements: HTMLInputElement[]): void {
            const validatorEmailMatchesConfig: ValidatorConfigs = {
                required: {
                    enabled: elements[0].hasAttribute("required"),
                },
                email: {},
                matches: {
                    id: elements[0].id,
                },
            };
            ValidationService.addValidatorsToElement(elements[1], validatorEmailMatchesConfig, true);
        },
    },
});
</script>

<template>
    <div>
        <f-text-field
            :id="id"
            type="email"
            :maxlength="maxLength"
            v-bind="$attrs"
            :model-value="modelValue"
            @change="onChange"
            @blur="onBlur"
            @update:model-value="onUpdate"
            @validity="onValidity"
            @pending-validity="onPendingValidity"
        >
            <!-- @slot Optional slot for label content. -->
            <slot name="default">{{ defaultText }}</slot>
            <template #error-message>
                <span v-if="showPasteErrorMessage">
                    {{ pasteErrorText }}
                </span>
            </template>
        </f-text-field>
        <f-text-field
            v-if="extendedValidation"
            v-model="secondEmail"
            type="email"
            :maxlength="maxLength"
            @paste="onPaste"
            @blur="showPasteErrorMessage = false"
        >
            <!-- @slot Optional slot for label content of extended field. -->
            <slot name="extended-label">{{ $t("fkui.email-text-field.label.repeat", "Upprepa mejladress") }}</slot>
        </f-text-field>
    </div>
</template>
