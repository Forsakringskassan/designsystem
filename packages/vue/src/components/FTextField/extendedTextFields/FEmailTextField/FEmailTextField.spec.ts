import "html-validate/jest";
import {
    ValidationService,
    type PendingValidityEvent,
    type ValidatableHTMLElement,
    type ValidityEvent,
    type ValidatorConfigs,
} from "@fkui/logic";
import { mount, VueWrapper } from "@vue/test-utils";
import flushPromises from "flush-promises";
import { defineComponent } from "vue";
import { createPlaceholderInDocument } from "@fkui/test-utils/vue";
import { ValidationPlugin } from "../../../../plugins";
import FEmailTextField from "./FEmailTextField.vue";

function createWrapper({
    props = {},
    slots = {},
    attrs = {},
    options = {},
} = {}): VueWrapper<InstanceType<typeof FEmailTextField>> {
    return mount(FEmailTextField, {
        attrs: { ...attrs },
        props: { maxLength: 80, ...props },
        slots: { ...slots },
        attachTo: createPlaceholderInDocument(),
        ...options,
        global: {
            stubs: ["f-icon"],
            plugins: [ValidationPlugin],
        },
    });
}

describe("snapshots", () => {
    it("should match snapshot with label and input", () => {
        const wrapper = createWrapper();

        expect(wrapper.element).toMatchSnapshot();
    });

    it("should match snapshot with label, error message and input", () => {
        const wrapper = createWrapper({
            slots: { "error-message": "ERROR_MESSAGE" },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("should match snapshot with label, tooltip, description, error message and input", () => {
        const wrapper = createWrapper({
            slots: {
                description: "DESCRIPTION",
                tooltip: "TOOLTIP",
                "error-message": "ERROR_MESSAGE",
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it.each`
        validityMode | isValid
        ${"VALID"}   | ${true}
        ${"ERROR"}   | ${false}
        ${"INITIAL"} | ${true}
        ${"INITIAL"} | ${false}
    `(
        "should match snapshot when validityMode is $validityMode and isValid is $isValid",
        async ({ validityMode, isValid }) => {
            const wrapper = createWrapper({
                attrs: { id: "elementId" },
            });

            const input = wrapper.get("input");
            const htmlInput = input.element;

            htmlInput.dispatchEvent(
                new CustomEvent<ValidityEvent>("validity", {
                    detail: {
                        target: wrapper.element as ValidatableHTMLElement,
                        elementId: "elementId",
                        isValid,
                        validityMode,
                        validationMessage: "Something went wrong.",
                        nativeEvent: "input",
                    },
                }),
            );
            await flushPromises();
            wrapper.vm.$forceUpdate();

            expect(wrapper.element).toMatchSnapshot();
        },
    );
});

describe("slots", () => {
    it("should support custom label in default and extended field", () => {
        const wrapper = createWrapper({
            props: {
                extendedValidation: true,
            },
            slots: {
                default: "Kontaktpersonens mejladress",
                "extended-label": "Upprepa kontaktpersonens mejladress",
            },
        });

        const labels = wrapper.findAll("label");
        const firstLabel = labels[0];
        const secondLabel = labels[1];

        expect(firstLabel.text()).toContain("Kontaktpersonens mejladress");
        expect(secondLabel.text()).toContain(
            "Upprepa kontaktpersonens mejladress",
        );
    });
});

describe("attributes", () => {
    it("should pass attributes", () => {
        const wrapper = createWrapper({
            attrs: {
                disabled: true,
                required: true,
            },
        });
        const input = wrapper.get("input");

        expect(input.attributes("disabled")).toBeDefined();
        expect(input.attributes("required")).toBeDefined();
    });

    it("should set type to email as default", () => {
        const wrapper = createWrapper();
        const input = wrapper.get("input");
        expect(input.attributes("type")).toBe("email");
    });
});

describe("events", () => {
    it("should support v-model", async () => {
        const wrapper = createWrapper({
            props: { modelValue: "foo@example.net" },
        });
        const input = wrapper.get("input");
        const htmlInput = input.element;
        expect(htmlInput.value).toBe("foo@example.net");

        input.setValue("bar@example.net");
        expect(htmlInput.value).toBe("bar@example.net");

        await input.trigger("change");
        expect(
            wrapper.emitted("update:modelValue")![0][0],
        ).toMatchInlineSnapshot(`"bar@example.net"`);
    });

    it("should pass listeners", async () => {
        const focus = jest.fn();
        const blur = jest.fn();

        const wrapper = createWrapper({
            attrs: {
                onFocus: focus,
                onBlur: blur,
            },
        });
        const input = wrapper.get("input");
        await input.trigger("focus");
        await wrapper.vm.$nextTick();
        await input.trigger("blur");
        await wrapper.vm.$nextTick();

        expect(focus).toHaveBeenCalled();
        expect(blur).toHaveBeenCalled();
    });

    it('should have ValidityMode INITIAL when "pending-validity" event is triggered', async () => {
        const wrapper = createWrapper({
            attrs: { id: "elementId" },
        });

        const input = wrapper.get("input");
        const htmlInput = input.element;

        htmlInput.dispatchEvent(
            new CustomEvent<ValidityEvent>("validity", {
                detail: {
                    target: wrapper.element as ValidatableHTMLElement,
                    elementId: "elementId",
                    isValid: true,
                    validityMode: "ERROR",
                    validationMessage: "Something went wrong.",
                    nativeEvent: "input",
                },
            }),
        );
        await flushPromises();
        wrapper.vm.$forceUpdate();

        expect(wrapper.vm.$data.validityMode).toBe("ERROR");

        htmlInput.dispatchEvent(
            new CustomEvent<PendingValidityEvent>("pending-validity"),
        );
        await flushPromises();
        wrapper.vm.$forceUpdate();

        expect(wrapper.vm.$data.validityMode).toBe("INITIAL");
    });
});

describe("validation", () => {
    it("should display correct error message when multiple validators", async () => {
        const wrapper = createWrapper({ options: { sync: false } });
        await flushPromises();
        ValidationService.setSubmitted(wrapper.element);

        const input = wrapper.get("input");
        const validatorConfigs: ValidatorConfigs = { required: {} };
        ValidationService.addValidatorsToElement(
            input.element,
            validatorConfigs,
        );

        await input.trigger("focus");
        input.setValue("");
        await input.trigger("blur");
        await flushPromises();

        expect(wrapper.get(".label__message--error").text()).toBe(
            "Fyll i en mejladress.",
        );

        await input.trigger("focus");
        input.setValue("5.5");
        await input.trigger("blur");
        await flushPromises();

        expect(wrapper.get(".label__message--error").text()).toBe(
            "Mejladressen Ã¤r inte korrekt ifylld.",
        );
    });

    it.each`
        required
        ${true}
        ${false}
    `(
        `should set confirm email textfield as required = $required when first email textfield is required = $required`,
        async ({ required }) => {
            const validation = required ? "v-validation.required" : "";
            const wrapper = mount(
                defineComponent({
                    name: "TestComponent",
                    components: {
                        FEmailTextField,
                    },
                    template: /* HTML */ `
                        <f-email-text-field ${validation} extendedValidation>
                        </f-email-text-field>
                    `,
                }),
                {
                    attachTo: createPlaceholderInDocument(),
                    global: {
                        plugins: [ValidationPlugin],
                    },
                },
            );

            const input = wrapper.findAll("input")[1];
            const expectedValue = required ? "" : undefined;
            expect(input.attributes("required")).toBe(expectedValue);
        },
    );
});

describe("disable paste", () => {
    function testPaste(inputElement: HTMLInputElement): void {
        const clipboardEvent = new Event("paste", {
            bubbles: true,
            cancelable: true,
            composed: true,
        });
        clipboardEvent.preventDefault = jest.fn();

        inputElement.dispatchEvent(clipboardEvent);
        expect(clipboardEvent.preventDefault).toHaveBeenCalled();
    }

    function allowPaste(inputElement: HTMLInputElement): void {
        const clipboardEvent = new Event("paste", {
            bubbles: true,
            cancelable: true,
            composed: true,
        });
        clipboardEvent.preventDefault = jest.fn();

        inputElement.dispatchEvent(clipboardEvent);
        expect(clipboardEvent.preventDefault).not.toHaveBeenCalled();
    }

    it("should paste content", async () => {
        const wrapper = createWrapper({ options: { sync: false } });
        expect(wrapper.find(".label__message--error").exists()).toBeFalsy();

        const inputElement = wrapper.get("input").element;
        allowPaste(inputElement);
        await flushPromises();

        expect(wrapper.find(".label__message--error").exists()).toBeFalsy();
    });

    it("should not paste content in second textfield", async () => {
        const wrapper = createWrapper({
            options: { sync: false },
            props: { extendedValidation: true },
        });
        expect(wrapper.find(".label__message--error").exists()).toBeFalsy();

        const secondInputElement = wrapper.findAll("input")[1].element;
        testPaste(secondInputElement);
        await flushPromises();

        expect(wrapper.get(".label__message--error").text()).toBe(
            "Du kan inte kopiera mejladressen. Du mÃ¥ste skriva in den igen.",
        );
    });

    it("should remove paste error message when field is validated", async () => {
        const wrapper = createWrapper({
            options: { sync: false },
            props: { extendedValidation: true },
        });
        const firstInputElement = wrapper.findAll("input")[0].element;
        const secondInputElement = wrapper.findAll("input")[1].element;

        allowPaste(firstInputElement);
        testPaste(secondInputElement);
        await flushPromises();

        expect(wrapper.get(".label__message--error").text()).toBe(
            "Du kan inte kopiera mejladressen. Du mÃ¥ste skriva in den igen.",
        );

        const input = wrapper.findAll("input")[1];
        await input.trigger("focus");
        input.setValue("bad.formatted.email");
        await input.trigger("blur");
        await flushPromises();

        expect(wrapper.get(".label__message--error").text()).toBe(
            "Mejladressen Ã¤r inte korrekt ifylld.",
        );
    });
});

describe("html-validate", () => {
    it.each`
        html
        ${'<f-email-text-field type="email" max-length="80">E-post</f-email-text-field>'}
        ${'<f-email-text-field id="email-input" max-length="80">E-post</f-email-text-field>'}
        ${'<f-email-text-field id="email-input" max-length="80" v-validation.required>E-post</f-email-text-field>'}
    `("$html should be valid", ({ html }) => {
        expect.assertions(1);

        expect(html).toHTMLValidate();
    });

    it("should allow custom label for default and extended field", () => {
        const markup = /* HTML */ `
            <f-email-text-field extended-validation>
                <template #default> E-post </template>
                <template #extended-label> Upprepa e-post </template>
            </f-email-text-field>
        `;
        expect(markup).toHTMLValidate();
    });
});
