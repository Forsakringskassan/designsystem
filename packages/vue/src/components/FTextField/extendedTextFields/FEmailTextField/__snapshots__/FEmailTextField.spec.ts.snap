// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should match snapshot when validityMode is ERROR and isValid is false 1`] = `
<div>
  <div
    class="text-field text-field--error"
  >
    <div
      class="i-width-sm-12"
    >
      <label
        class="label"
        for="elementId"
      >
        
        
        
        Mejladress
        
        
        
        
        
        <span
          class="label__message label__message--error"
        >
          <f-icon-stub
            class="label__icon--left"
            library="f"
            name="error"
          />
          
          
          
          Something went wrong.
          
          
          
        </span>
      </label>
    </div>
    <div
      class="text-field__input-wrapper i-width-sm-12"
    >
      
      
      <div
        class="text-field__icon-wrapper"
      >
        <input
          class="text-field__input"
          data-validation=""
          id="elementId"
          maxlength="80"
          type="email"
        />
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <!--v-if-->
  </div>
  <!--v-if-->
</div>
`;

exports[`snapshots should match snapshot when validityMode is INITIAL and isValid is false 1`] = `
<div>
  <div
    class="text-field"
  >
    <div
      class="i-width-sm-12"
    >
      <label
        class="label"
        for="elementId"
      >
        
        
        
        Mejladress
        
        
        
        
        
        <!--v-if-->
      </label>
    </div>
    <div
      class="text-field__input-wrapper i-width-sm-12"
    >
      
      
      <div
        class="text-field__icon-wrapper"
      >
        <input
          class="text-field__input"
          data-validation=""
          id="elementId"
          maxlength="80"
          type="email"
        />
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <!--v-if-->
  </div>
  <!--v-if-->
</div>
`;

exports[`snapshots should match snapshot when validityMode is INITIAL and isValid is true 1`] = `
<div>
  <div
    class="text-field"
  >
    <div
      class="i-width-sm-12"
    >
      <label
        class="label"
        for="elementId"
      >
        
        
        
        Mejladress
        
        
        
        
        
        <!--v-if-->
      </label>
    </div>
    <div
      class="text-field__input-wrapper i-width-sm-12"
    >
      
      
      <div
        class="text-field__icon-wrapper"
      >
        <input
          class="text-field__input"
          data-validation=""
          id="elementId"
          maxlength="80"
          type="email"
        />
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <!--v-if-->
  </div>
  <!--v-if-->
</div>
`;

exports[`snapshots should match snapshot when validityMode is VALID and isValid is true 1`] = `
<div>
  <div
    class="text-field"
  >
    <div
      class="i-width-sm-12"
    >
      <label
        class="label"
        for="elementId"
      >
        
        
        
        Mejladress
        
        
        
        
        
        <!--v-if-->
      </label>
    </div>
    <div
      class="text-field__input-wrapper i-width-sm-12"
    >
      
      
      <div
        class="text-field__icon-wrapper"
      >
        <input
          class="text-field__input"
          data-validation=""
          id="elementId"
          maxlength="80"
          type="email"
        />
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <!--v-if-->
  </div>
  <!--v-if-->
</div>
`;

exports[`snapshots should match snapshot with label and input 1`] = `
<div>
  <div
    class="text-field"
  >
    <div
      class="i-width-sm-12"
    >
      <label
        class="label"
        for="fkui-vue-element-0001"
      >
        
        
        
        Mejladress
        
        
        
        
        
        <!--v-if-->
      </label>
    </div>
    <div
      class="text-field__input-wrapper i-width-sm-12"
    >
      
      
      <div
        class="text-field__icon-wrapper"
      >
        <input
          class="text-field__input"
          data-validation=""
          id="fkui-vue-element-0001"
          maxlength="80"
          type="email"
        />
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <!--v-if-->
  </div>
  <!--v-if-->
</div>
`;

exports[`snapshots should match snapshot with label, error message and input 1`] = `
<div>
  <div
    class="text-field"
  >
    <div
      class="i-width-sm-12"
    >
      <label
        class="label"
        for="fkui-vue-element-0001"
      >
        
        
        
        Mejladress
        
        
        
        
        
        <!--v-if-->
      </label>
    </div>
    <div
      class="text-field__input-wrapper i-width-sm-12"
    >
      
      
      <div
        class="text-field__icon-wrapper"
      >
        <input
          class="text-field__input"
          data-validation=""
          id="fkui-vue-element-0001"
          maxlength="80"
          type="email"
        />
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <!--v-if-->
  </div>
  <!--v-if-->
</div>
`;

exports[`snapshots should match snapshot with label, tooltip, description, error message and input 1`] = `
<div>
  <div
    class="text-field"
  >
    <div
      class="i-width-sm-12"
    >
      <label
        class="label"
        for="fkui-vue-element-0001"
      >
        
        
        
        Mejladress
        
        
        
        
        
        <!--v-if-->
      </label>
    </div>
    <div
      class="text-field__input-wrapper i-width-sm-12"
    >
      
      
      <div
        class="text-field__icon-wrapper"
      >
        <input
          class="text-field__input"
          data-validation=""
          id="fkui-vue-element-0001"
          maxlength="80"
          type="email"
        />
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <!--v-if-->
  </div>
  <!--v-if-->
</div>
`;
